File: Backend/Microservices/Admin/Eefa.Admin.Application.Refactor/UseCases/PersonBankAccounts/Commands/Delete/DeletePersonBankAccountCommand.cs
using AutoMapper;
using MediatR;
using System.Threading;
using System.Threading.Tasks;
using Eefa.Admin.Application.CommandQueries.PersonBankAccounts.Models;

public class DeletePersonBankAccountCommand : IRequest<ServiceResult<PersonBankAccountModel>>, IMapFrom<PersonBankAccount>
{
    public int Id { get; set; }

    public void Mapping(Profile profile)
    {
        profile.CreateMap<DeletePersonBankAccountCommand, PersonBankAccount>();
    }
}

public class DeletePersonBankAccountCommandHandler : IRequestHandler<DeletePersonBankAccountCommand, ServiceResult<PersonBankAccountModel>>
{
    private readonly IUnitOfWork _unitOfWork;
    private readonly IMapper _mapper;

    public DeletePersonBankAccountCommandHandler(IUnitOfWork unitOfWork, IMapper mapper)
    {
        _unitOfWork = unitOfWork;
        _mapper = mapper;
    }
    public async Task<ServiceResult<PersonBankAccountModel>> Handle(DeletePersonBankAccountCommand request, CancellationToken cancellationToken)
    {
        PersonBankAccount entity = await _unitOfWork.PersonBankAccounts.GetByIdAsync(request.Id);

        _unitOfWork.PersonBankAccounts.Delete(entity);
        await _unitOfWork.SaveChangesAsync(cancellationToken);

        return ServiceResult.Success(_mapper.Map<PersonBankAccountModel>(entity));
    }
}

File: Frontend/angular-app/src/app/modules/admin/repositories/person-bank-accounts/commands/delete-person-bank-account-command.ts
import {HttpRequest} from "../../../../../core/services/http/http-request";
import {Inject} from "@angular/core";
import {ServiceResult} from "../../../../../core/models/service-result";
import {IRequest, IRequestHandler} from "../../../../../core/services/mediator/interfaces";
import {PersonBankAccount} from "../../../entities/person-bank-account";
import {HttpService} from "../../../../../core/services/http/http.service";
import {ApplicationError} from "../../../../../core/exceptions/application-error";
import {MediatorHandler} from "../../../../../core/services/mediator/decorator";
import {ValidationRule} from "../../../../../core/validation/validation-rule";
import {NotificationService} from "../../../../../shared/services/notification/notification.service";

export class DeletePersonBankAccountCommand extends IRequest<DeletePersonBankAccountCommand, PersonBankAccount> {

  constructor(public entityId:number) {
    super();
  }

  mapFrom(entity: PersonBankAccount): DeletePersonBankAccountCommand {
    throw new ApplicationError(DeletePersonBankAccountCommand.name, this.mapTo.name, 'Not Implemented Yet')
  }

  mapTo(): PersonBankAccount {
    throw new ApplicationError(DeletePersonBankAccountCommand.name, this.mapTo.name, 'Not Implemented Yet')
  }

  get url(): string {
    return "/admin/personBankAccounts/delete";
  }

  get validationRules(): ValidationRule[] {
    return [];
  }
}

@MediatorHandler(DeletePersonBankAccountCommandHandler.name)
export class DeletePersonBankAccountCommandHandler implements IRequestHandler<DeletePersonBankAccountCommand, PersonBankAccount> {
  constructor(
    @Inject(HttpService) private _httpService: HttpService,
    @Inject(NotificationService) private _notificationService: NotificationService,

  ) {
  }

  async Handle(request: DeletePersonBankAccountCommand): Promise<PersonBankAccount> {
    let httpRequest: HttpRequest<DeletePersonBankAccountCommand> = new HttpRequest<DeletePersonBankAccountCommand>(request.url, request);
    httpRequest.Query  += `id=${request.entityId}`;

    return await this._httpService.Delete<ServiceResult<PersonBankAccount>>(httpRequest).toPromise().then(response => {
      this._notificationService.showSuccessMessage()

      return response.objResult
    })
  }
}
